### PR TITLE
Refactor factory and remove unnecessary constraint + increase file size limit

### PR DESCRIPTION
**Pull request by bartcharbon**

### CODE REVIEW

File: gene-network/src/main/resources/js/gn.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ $(function () {
             },
             validExtensions: ['.vcf', '.vcf.gz', '.tsv', '.tsv.gz', '.txt', '.txt.gz', '.tab', 'tab.gz'],
             showNameFieldExtensions: ['.vcf', '.vcf.gz', '.tsv', '.tsv.gz', '.txt', '.txt.gz', '.tab', 'tab.gz'],
-            maxFileSizeMB: 3
+            maxFileSizeMB: 50
```
Comment (ConnorStroomberg) [inline gene-network/src/main/resources/js/gn.js]: Why 50 ? , maybe add comment to explain why 50 is a good 'maxFileSize'

File: gene-network/src/main/java/org/molgenis/ui/genenetwork/matrix/meta/MatrixMetadata.java [hunk 2]
```diff
@@ -47,7 +47,7 @@ public void init()
 		setLabel("Matrix metadata");
 		setDescription("metadata with information about the matrix file");
 		addAttribute(ID, ROLE_ID);
-		addAttribute(FILE_LOCATION, ROLE_LABEL, ROLE_LOOKUP).setLabel("Location of the matrix file").setUnique(true)
```
Comment (ConnorStroomberg) [inline gene-network/src/main/java/org/molgenis/ui/genenetwork/matrix/meta/MatrixMetadata.java]: Why is this not unique ?

Comment (bartcharbon) [inline gene-network/src/main/java/org/molgenis/ui/genenetwork/matrix/meta/MatrixMetadata.java]: You might want to use the matrix with different (or without) row and column mappings.

File: gene-network/src/main/resources/js/gn.js (hunk 1 above)
Comment (bartcharbon) [inline gene-network/src/main/resources/js/gn.js]: I'll add a comment, but it remains a bit arbitrary. It's an estimate of the filesize of an exome VCF

**Pull request merged**